File: capacity-club-api/src/domain-modules/address/model/entity/address.entity.ts
import { UniqueId, uniqueIdTransformer } from '@common/model/unique-id';
import { Transform } from 'class-transformer';
import { Entity, BaseEntity, PrimaryColumn, Column } from 'typeorm';

/**
 * Entity representing an address.
 */
@Entity()
export class Address extends BaseEntity {
  /**
   * Unique identifier for the address.
   * Uses a custom transformer for serialization.
   */
  @PrimaryColumn('varchar')
  @Transform(uniqueIdTransformer.to, { toClassOnly: true })
  @Transform(uniqueIdTransformer.from, { toPlainOnly: true })
  address_id: UniqueId;
  @Column({ length: 50, nullable: false, unique: false })
  street: string;
  @Column({ length: 8, nullable: false, unique: false })
  number: string;
  @Column({ length: 10, nullable: false, unique: false })
  zipcode: string;
  @Column({ length: 50, nullable: false, unique: false })
  town: string;
  @Column({ length: 50, nullable: false, unique: false })
  country: string;
  @Column({ nullable: true, unique: false })
  complement: string;
}

File: capacity_club_mobile_app/lib/domain-features/domain/address/entity/address_entity.dart
import 'package:capacity_club_mobile_app/core/model/entities/unique_id.dart';

class AddressEntity {
  final UniqueId address_id;
  final String street;
  final String number;
  final String zipcode;
  final String town;
  final String country;
  final String complement;
  final bool is_empty;

  AddressEntity(
      {required this.address_id,
      required this.street,
      required this.number,
      required this.zipcode,
      required this.town,
      required this.country,
      required this.complement,
      required this.is_empty});

  factory AddressEntity.empty() {
    return AddressEntity(
        address_id: UniqueId(''),
        street: '',
        number: '',
        zipcode: '',
        town: '',
        country: '',
        complement: '',
        is_empty: true);
  }
}
